src/js/jsx/shared/Dialog.jsx

```diff
@@ -124,11 +132,10 @@ define(function (require, exports, module) {
                 id = this.props.id;
 
             if (this.state.open) {
+                this._target = null;
                 flux.actions.dialog.closeDialog(id);
             } else if (!this.props.disabled) {
-                this.setState({
-                    target: event.target
-                });
+                this._target = event.target;
```

Comment:
would it also have been possible to call the `openDialog` action in the callback of setState?